forest/shared/src/cli/snapshot_fetch.rs

```diff
@@ -650,23 +652,52 @@ mod test {
             return Ok(());
         }
 
-        let (url, shutdown_tx) = serve_forest_logo()?;
+        let (url, shutdown_tx, shasum, _data_dir) = serve_random_file()?;
         download_with_aria2(
             &url,
             temp_dir().as_os_str().to_str().unwrap_or_default(),
             "test",
-            "sha-256=d06d7f5613640befb07f99079ffcbdc55fff064fc1dfa8041acf2547f5fa1a6e",
+            &format!("sha-256={}", hex::encode(shasum)),
         )?;
         shutdown_tx.send(()).unwrap();
         Ok(())
     }
 
-    fn serve_forest_logo() -> Result<(String, tokio::sync::oneshot::Sender<()>)> {
+    /// Serves a random file over HTTP.
+    /// Returns:
+    /// - url of the served file,
+    /// - service channel,
+    /// - expected SHA-256 of the file,
+    /// - handle to the temporary directory in which the file is created.
+    fn serve_random_file() -> Result<(
+        String,
+        tokio::sync::oneshot::Sender<()>,
+        sha2::digest::Output<Sha256>,
+        TempDir,
```

Comment:
loads of items in the tuple, but i guess it's fine given these are only for tests.